libreoffice-core/desktop/source/lib/init.cxx

```diff
@@ -889,6 +892,49 @@ void ExecuteMarginULChange(
                                                           SfxCallMode::RECORD, { pPageULMarginItem });
 }
 
+void setPageSize(
+    const tools::Long Width,
+    const tools::Long Height
+)
+{
+    SfxViewFrame* pViewFrm = SfxViewFrame::Current();
+
+    if (!pViewFrm)
+        return;
+
+
+    SfxDispatcher* pDispatcher = pViewFrm->GetDispatcher();
+    const SvxSizeItem* pSizeItem = new SvxSizeItem( SID_ATTR_PAGE_SIZE, Size(Width, Height));
+
+    pDispatcher->ExecuteList(SID_ATTR_PAGE_SIZE, SfxCallMode::RECORD, { pSizeItem });
+}
+
+
+void setPageMargins(
+    const tools::Long PageLeft,
+    const tools::Long PageRight,
+    const tools::Long PageTop,
+    const tools::Long PageBottom
+)
+{
+    SfxViewFrame* pViewFrm = SfxViewFrame::Current();
+
+    if (!pViewFrm)
+        return;
+
+    SvxLongLRSpaceItem* pLRSpaceItem = new SvxLongLRSpaceItem();
+    pLRSpaceItem->SetLeft(PageLeft);
+    pLRSpaceItem->SetRight(PageRight);
```

Comment:
Best way to validate this is to is correct is to save the document, and pop it into LibreOffice and seeing if the values are the right ones